python310Packages.pygls: 0.13.0 -> 1.0.0, python310Packges.cmake-language-server: 0.1.6 -> unstable-2023-01-08

fabaff opened the pull request:
###### Description of changes
Fixes #211215 (https://hydra.nixos.org/build/205594188)

Based on the work of @doronbehar and @B4dM4n in #204457

Required for #211881
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

doronbehar committed:
python310Packages.pygls: 0.13.0 -> 1.0.0

Diff: https://github.com/openlawlibrary/pygls/compare/v0.13.0...v1.0.0

Changelog: https://github.com/openlawlibrary/pygls/blob/v1.0.0/CHANGELOG.md

doronbehar committed:
python310Packages.nox: init at 2022.11.21

doronbehar committed:
python310Packages.lsprotocol: init at 2022.0.0a9

fabaff committed:
python310Packages.lsprotocol: adjust meta

- disable on unsupported Python releases

fabaff commented:
Result of `nixpkgs-review pr 211901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>cmake-language-server</li>
    <li>python310Packages.jedi-language-server</li>
    <li>python310Packages.lsprotocol</li>
    <li>python310Packages.nox</li>
    <li>python310Packages.pygls</li>
    <li>python311Packages.jedi-language-server</li>
    <li>python311Packages.lsprotocol</li>
    <li>python311Packages.nox</li>
    <li>python311Packages.pygls</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vscode-extensions.ms-python.python</li>
    <li>ycmd</li>
  </ul>
</details>

kira-bruneau commented:
Result of `nixpkgs-review pr 211901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>cmake-language-server</li>
    <li>python310Packages.jedi-language-server</li>
    <li>python310Packages.lsprotocol</li>
    <li>python310Packages.nox</li>
    <li>python310Packages.pygls</li>
    <li>python311Packages.jedi-language-server</li>
    <li>python311Packages.lsprotocol</li>
    <li>python311Packages.nox</li>
    <li>python311Packages.pygls</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vscode-extensions.ms-python.python</li>
    <li>ycmd</li>
  </ul>
</details>

kira-bruneau reviewed (commented):
Thanks for all these fixes @fabaff!! I noticed that cmake-language-server is giving back the wrong version now though: 

```
> cmake-language-server --version
cmake-language-server 0.0.0
```

~I can take a look at this though!~ - EDIT: I found out we can just set this for now:

```nix
  PDM_PEP517_SCM_VERSION = "2013.1";
```

kira-bruneau commented:
Oh also, #211881 doesn't depend on this PR - debugpy is a dap server, not an lsp server and doesn't depend on pygls or lsprotocol.

kira-bruneau commented on pkgs/development/tools/misc/cmake-language-server/default.nix:
```diff
@@ -1,39 +1,33 @@
     # Test timeouts occasionally cause the build to fail
     ./disable-test-timeouts.patch
   ];
 
```
Hmm it looks like we can just add this to fix the cmake-language-server version:

```suggestion

  PDM_PEP517_SCM_VERSION = version;
```

but I think it only takes semver-style versions, this is what I get with the change:

```
> cmake-language-server --version
cmake-language-server 8
```

I wonder if we should just preemptively set this to 0.1.7?

kira-bruneau replied:
Oh it actually uses [PEP 440](https://peps.python.org/pep-0440/), so I think we could then just set it as:

```suggestion

  PDM_PEP517_SCM_VERSION = "2013.1";
```

fabaff replied:
Added, thanks.

fabaff committed:
python310Packages.jedi-language-server: add missing inputs

- disable on unsupported Python releases

fabaff committed:
python310Packages.nox: fix build

- add changelog to meta
- adjust inputs
- disable on unsupported Python releases

fabaff commented on pkgs/development/python-modules/jedi-language-server/default.nix:
```diff
@@ -1,27 +1,32 @@
-, pythonRelaxDepsHook
 , docstring-to-markdown
+, fetchFromGitHub
+, fetchPypi
```
Removed, thanks.

B4dM4n commented on pkgs/development/python-modules/lsprotocol/default.nix:
```diff
@@ -0,0 +1,63 @@
+    cattrs
+  ];
+
+  checkInputs = [
+    jsonschema
+    pyhamcrest
+    pytest
+  ];
```
The `lsprotocol` build is failing with `nox > Program pytest not found.` when running `nixpkgs-review`.

This seems to happen because of  #206742. It has to use `nativeCheckInputs` for `pytest` now. 

```suggestion
  nativeCheckInputs = [
    pytest
  ];

  checkInputs = [
    jsonschema
    pyhamcrest
  ];
```

fabaff replied:
Thanks, updated.

fabaff committed:
python310Packages.lsprotocol: switch to nativeCheckInputs

fabaff commented:
Result of `nixpkgs-review pr 211901` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>cmake-language-server</li>
    <li>python310Packages.jedi-language-server</li>
    <li>python310Packages.lsprotocol</li>
    <li>python310Packages.nox</li>
    <li>python310Packages.pygls</li>
    <li>python311Packages.jedi-language-server</li>
    <li>python311Packages.lsprotocol</li>
    <li>python311Packages.nox</li>
    <li>python311Packages.pygls</li>
    <li>vimPlugins.YouCompleteMe</li>
    <li>vscode-extensions.ms-python.python</li>
    <li>ycmd</li>
  </ul>
</details>

B4dM4n commented on pkgs/development/tools/misc/cmake-language-server/default.nix:
```diff
@@ -1,39 +1,35 @@
+    hash = "sha256-vNG43sZy2wMetY5mbgxIoei5jCCj1f8vWiovWtwzbPc=";
   };
 
+  PDM_PEP517_SCM_VERSION = "2013.1";
```
I guess there is a typo here. It should probably say:
```suggestion
  PDM_PEP517_SCM_VERSION = "2023.1";
```

fabaff replied:
Thanks for checking. Changed.

B4dM4n reviewed (approved):
One minor note as inline comment, otherwise it looks great.

fabaff committed:
python310Packges.cmake-language-server: 0.1.6 -> unstable-2023-01-08
